framework/sandstone_tests.h

```diff
@@ -52,23 +61,20 @@ class SandstoneTestSet
 public:
     using TestSet = std::vector<struct test *>;
 
-    static TestSet lookup(const char *name);
-
-    SandstoneTestSet() : SandstoneTestSet(false) {
-    };
-
     /* searches the test by name in all the available tests. */
     TestSet lookup(const char *name);
 
     SandstoneTestSet(struct test_set_cfg cfg) : SandstoneTestSet(cfg, false) {
     };
 
-    SandstoneTestSet(bool all_tests, bool is_selftest);
+    SandstoneTestSet(struct test_set_cfg cfg, bool all_tests) : SandstoneTestSet(cfg, true, false) {
+    };
+
+    SandstoneTestSet(struct test_set_cfg cfg, bool all_tests, bool is_selftest);
```

Comment:
Either merge the `all_tests` and `is_selftest` booleans into the config or move the three new booleans into the flags enum that I suggested earlier.